benchmarks/synth-bm/Cargo.toml

```diff
@@ -1,23 +1,38 @@
 [package]
 name = "near-synth-bm"
-version = "0.0.0"
-authors = ["Near Inc <hello@nearprotocol.com>"]
-edition = "2021"
-rust-version = "1.84.0"
-repository = "https://github.com/near/nearcore"
-license = "MIT OR Apache-2.0"
+version.workspace = true
+authors.workspace = true
+edition.workspace = true
+rust-version.workspace = true
+repository.workspace = true
+license.workspace = true
+publish = false
+
+[lints]
+workspace = true
+
 
 [dependencies]
-anyhow = "1.0.89"
-clap = { version = "4.5.17", features = ["derive"] }
-env_logger = "0.11.5"
-log = "0.4.22"
+anyhow.workspace = true 
+clap = { workspace = true, features = ["derive"] }
+env_logger.workspace = true
+log.workspace = true
+tokio = { workspace = true, features = ["rt", "rt-multi-thread", "macros"] }
+serde = { workspace = true, features = ["derive"] }
+serde_json.workspace = true
+tracing = { workspace = true, features = ["std"] }
+rand.workspace = true
+
+# There are two issues related to inheriting below dependencies from the root workspace:
+#
+# 1) There is an unpublished workspace member in `nearcore/chain/jsonrpc/client` whose name
+#    collides with the crate `near-jsonrpc-client` that is published on crates.io and lives in
+#    another repo. Many workspace members already use the local `near-jsonrpc-client` crate.
+# 2) `near-jsonrpc-client` (from crates.io) restricts the versions of other `near-*` crates it
+#    is compatible with. These can currently not be satisfied by path dependencies: for example
+#    `near-crypto.workspace` would pull in a `near-crypto` version that differs from the one used by
+#    `near-jsonrpc-client`, leading to compilation errors.
```

Comment:
@nagisa given these issues, do you think it is reasonable to *not* inherit the `near-*` dependencies below from the workspace?

Reply:
I think its fine to not inherit these dependencies. At the same time I believe it is a bug that `near-jsonrpc-client` has `near-primitives` and(?) `near-crypto` in their public interface. The `jsonrpc-client` should be re-exporting the types and/or the crate it exposes as a public interface instead. Otherwise we're going to play version catchup with this crate all the time anyway, or more likely, just leave it to bitrot.

cc @akorchyn

Reply:
The `near-jsonrpc-client` in our tree is used by some other crates in the workspace, so I would do the renaming in a separate PR. Opened #12908.

Reply:
@mooori @nagisa Just for your information, I have created the following tracking issue for a more general problem with the RPC clients: #12939